01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js

```diff
@@ -1,14 +1,123 @@
+  ' "father": "Vincent de Pape", "mother": "Petronella Wauters"}',
+  '{"name": "Daniel Haverbeke", "sex": "m", "born": 1652, "died": 1723,' +
+  ' "father": "Lieven Haverbeke", "mother": "Elisabeth Hercke"}',
+  '{"name": "Lieven Haverbeke", "sex": "m", "born": 1631, "died": 1676,' +
+  ' "father": "Pieter Haverbeke", "mother": "Anna van Hecke"}',
+  '{"name": "Martina de Pape", "sex": "f", "born": 1666, "died": 1727,' +
+  ' "father": "Vincent de Pape", "mother": "Petronella Wauters"}',
+  '{"name": "Jan Francies Haverbeke", "sex": "m", "born": 1725,' +
+  ' "died": 1779,' +
+  ' "father": "Pieter Bernard Haverbeke", "mother": "Livina de Vrieze"}',
+  '{"name": "Maria Haverbeke", "sex": "m", "born": 1905, "died": 1997,' +
+  ' "father": "Emile Haverbeke", "mother": "Emma de Milliano"}',
+  '{"name": "Petronella de Decker", "sex": "f", "born": 1731, "died": 1781,' +
+  ' "father": "Pieter de Decker", "mother": "Livina Haverbeke"}',
+  '{"name": "Livina Sierens", "sex": "f", "born": 1761, "died": 1826,' +
+  ' "father": "Jan Sierens", "mother": "Maria van Waes"}',
+  '{"name": "Laurentia Haverbeke", "sex": "f", "born": 1710, "died": 1786,' +
+  ' "father": "Jan Haverbeke", "mother": "Maria de Rycke"}',
+  '{"name": "Carel Haverbeke", "sex": "m", "born": 1796, "died": 1837,' +
+  ' "father": "Pieter Antone Haverbeke", "mother": "Livina Sierens"}',
+  '{"name": "Elisabeth Hercke", "sex": "f", "born": 1632, "died": 1674,' +
+  ' "father": "Willem Hercke", "mother": "Margriet de Brabander"}',
+  '{"name": "Jan Haverbeke", "sex": "m", "born": 1671, "died": 1731,' +
+  ' "father": "Lieven Haverbeke", "mother": "Elisabeth Hercke"}',
+  '{"name": "Anna van Hecke", "sex": "f", "born": 1607, "died": 1670,' +
+  ' "father": "Paschasius van Hecke", "mother": "Martijntken Beelaert"}',
+  '{"name": "Maria Sturm", "sex": "f", "born": 1835, "died": 1917,' +
+  ' "father": "Charles Sturm", "mother": "Seraphina Spelier"}',
+  '{"name": "Jacobus Bernardus van Brussel", "sex": "m", "born": 1736,' +
+  ' "died": 1809,' +
+  ' "father": "Jan van Brussel",' +
+  ' "mother": "Elisabeth Haverbeke"}'].join(',\n  ') + '\n]';
+
+var ancestry = JSON.parse(ANCESTRY_FILE);
+
 function average(array) {
   function plus(a, b) {
     return a + b;
   }
   return array.reduce(plus) / array.length;
 }
 
-console.log(average([2, 2, 2]));
-// → 2
+function groupByCentury(arr) {
+  var century;
+  var result = {};
+  arr.map(function(el) {
+    century = Math.ceil(el.died / 100);
+    if (result.hasOwnProperty(century)) {
+      result[century].push(el.died - el.born);
+    } else {
+      result[century] = [el.died - el.born];
+    }
+    return null;
+  });
+  return result;
+}
+
+function printByCentury(arr) {
+  var key;
+  for (key in arr) {
+    if (arr.hasOwnProperty(key)) {
+      console.log(key + ': ' + average(arr[key]).toFixed(1));
+    }
+  }
+}
 
-// → Your code here.
+console.log(printByCentury(groupByCentury(ancestry)));
```

Comment:
Too many functions hide all under 

`lifeExpectancy` fuction + you may need [such header](https://github.com/nesterone/sf-eng-init-gl-spring-2016/blob/master/01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js#L1)